crypto/pem/pem_pk8.c

```diff
@@ -74,8 +74,8 @@ static int do_pk8pkey(BIO *bp, const EVP_PKEY *x, int isder, int nid,
     int ret = 0;
     const char *outtype = isder ? "DER" : "PEM";
     OSSL_ENCODER_CTX *ctx =
-        OSSL_ENCODER_CTX_new_by_EVP_PKEY(x, outtype, OSSL_KEYMGMT_SELECT_ALL,
-                                         libctx, propq);
+        OSSL_ENCODER_CTX_new_by_EVP_PKEY(x, OSSL_KEYMGMT_SELECT_ALL,
+                                         outtype, "pkcs8", libctx, propq);
```

Comment:
Not sure I like these hardwired strings like "pkcs8" being littered everywhere.

Reply:
I was thinking about that, but was unsure where to add it.  Should there be a set of macros in `include/openssl/encoder.h`, `include/openssl/evp.h` or `include/openssl/asn1.h`?  And re littering, how do you like them "DER" and "PEM"?

Since the structure names are fairly closely related to EVP_PKEY (because that's different output structures to apply on keys), I would rather go for `include/openssl/evp.h`.  Macro name suggestions?

Reply:
You have a list of macro names in encoder.inc currently..
they are encoder related so maybe that is a better place? OSSL_ENCODER_TYPE_XXX??

"type-specific"
"dh"
"dhx"
"dsa"
"ec"
"rsa"
"pkcs1"
"pkcs3"
"X9.42"
"X9.62"

Coming up with names is hard.

Reply:
They are also key related.  OSSL_ENCODER (and OSSL_DECODER) is designed for a future where other OpenSSL types are handled by providers as well.  "pkcs1" would not at all make sense to encode an X509, for example.

Reply:
(output structures aren't even relevant for a X509)

Reply:
It's quite possible that the answer is to have NAME macros for them in `include/openssl/core_names.h`.  So macros like this?

#define OSSL_STRUCTURE_NAME_PKCS1 "pkcs1"

Reply:
I am not sold on the 'structure name'.. Is that really what it is?

Reply:
Yes!  For encoding, that's exactly what it is.  PKCS#8 defines an ASN.1 structure.  PKCS#1 defines an ASN.1 structure.  SubjectPublicKeyInfo is an ASN.1 structure.

Those are, for encoding (and decoding) a name that indicates how the encoded data is organized (i.e. structured)

Reply:
Also keep in mind that "DER", "PEM" and "TEXT" are caller output *types* already...  I'm trying hard to have a terminology that doesn't exasperate confusion.